suites/apis/googleServiceAccountRemovalTest.js

```diff
@@ -0,0 +1,471 @@
+
+  // Lock the project
+  lockRes = await google.lockGoogleProject(googleProject);
+  chai.expect(lockRes, 'Could not lock project').to.be.true;
+
+  // Get creds to access data
+  let [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+
+  // Access data
+  user0AccessQARes = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Delete the creds so that the clean up job will not mark this SA as invalid
+  await google.deleteServiceAccountKey(keyFullName);
+  files.deleteFile(pathToKeyFile);
+
+  // Run clean up job
+  checkAndCleanSA();
+
+  // Get creds to access data
+  [pathToKeyFile, keyFullName] = await google.createServiceAccountKeyFile(googleProject);
+
+  // Access data
+  user0AccessQAResAfter = await google.getFileFromBucket(
+    fence.props.googleBucketInfo.QA.googleProjectId,
+    pathToKeyFile,
+    fence.props.googleBucketInfo.QA.bucketId,
+    fence.props.googleBucketInfo.QA.fileName
+  );
+
+  // Clean up
+  console.log('cleaning up');
+
+  await google.deleteServiceAccountKey(keyFullName);
+  files.deleteFile(pathToKeyFile);
+
+  await fence.do.deleteGoogleServiceAccount(
+    users.user0,
+    googleProject.serviceAccountEmail,
+  );
+
+  await fence.do.unlinkGoogleAcct(
+    users.user0,
+  );
+
+  // Unlock the project
+  let unlockRes = await google.unlockGoogleProject(googleProject);
+
+  // Asserts
+  chai.expect(user0AccessQARes,
+    'User should have bucket access before clean up job'
+  ).to.have.property('id');
+
+  chai.expect(user0AccessQAResAfter,
+    'User should have bucket access after clean up job'
+  ).to.have.property('id');
+
+  chai.expect(unlockRes, 'Could not unlock project').to.be.true;
+}).retry(2);
+
+
+// TODO: enable. this test fails until the jenkins envs are setup with the new monitoring SA
+xScenario('SA removal job test: monitor SA does not have access @reqGoogle', async (fence, users, google, files) => {
+  // test invalid SA because the monitor does not have access
+
+  // Setup
+  const googleProject = fence.props.googleProjectDynamic;
+  let getMonitorRes = await fence.do.getGoogleServiceAccountMonitor(users.user0);
+  fence.ask.assertStatusCode(getMonitorRes, 200, 'Could not get SA monitor info');
+  let monitorEmail = getMonitorRes.body.service_account_email;
+
+  await fence.complete.forceLinkGoogleAcct(users.user0, googleProject.owner);
+
+  // Register account
+  registerRes = await fence.do.registerGoogleServiceAccount(
+    users.user0,
+    googleProject,
+    ['QA']
+  );
+  fence.ask.responsesEqual(registerRes, fence.props.resRegisterServiceAccountSuccess);
+
+  // Lock the project
+  lockRes = await google.lockGoogleProject(googleProject);
+  chai.expect(lockRes, 'Could not lock project').to.be.true;
```

Comment:
can this be a bit more specific of an error message in case someone isn't familiar with what it tried to do and how it could fail? I'd rather have a paragraph here explaining then have someone bug you later about what this means. so like: `Could not lock Google project to run tests where we modifying the project iteself. Locks are necessary to avoid interfering with other testing envs since they all use this Google Project: {{name}}. We attempted to "lock" the env by creating a service account unique to this test run. We either could not create the SA or {{insert other potential failures here}}, the tests tried multiple times. Perhaps check the logs for Google API responses.`

Reply:
I know it seems excessive but it'll really help someone in the future if this randomly doesn't work

Reply:
something similar for `unlock` failures too.. possibly with more info about how to remove the stray locking SA if it fails

Reply:
done in 61b67026498d7a8503fc5d0e85c47c242ed5ac10